src/commands/flutter-symbols/upload.ts

```diff
@@ -117,10 +122,26 @@ export class UploadCommand extends Command {
 
     this.context.stdout.write(renderCommandInfo(this.dryRun, this.version!, this.serviceName, this.flavor, uploadInfo))
 
-    this.config = await resolveConfigFromFile(this.config, {
-      configPath: this.configPath,
-      defaultConfigPaths: DEFAULT_CONFIG_PATHS,
-    })
+    this.config = await resolveConfigFromFileAndEnvironment(
+      this.config,
+      {
+        apiKey: process.env.DATADOG_API_KEY,
+        datadogSite: process.env.DATADOG_SITE,
+      },
+      {
+        configPath: this.configPath,
+        defaultConfigPaths: DEFAULT_CONFIG_PATHS,
+        configFromFileCallback: (configFromFile: any) => {
```

Comment:
I feel weird having this callback duplicated in 3 places, as well as the error message. Is there any clean way to consolidate them into the main function?

Reply:
I'm open to merging these 3 occurences, but I would need some help on how to keep the current separation of concerns.

For each command, the output messages are formatted by a file called `renderer.ts`, and the reference to the `stdout` is kept in the `execute` method. 
I'm not really into passing either the `renderDuplicateAPIKey` or `stdout` to the `resolveConfigFromFileAndEnvironment` util function.

Speaking of the `resolveConfigFromFileAndEnvironment` function, I did not feel right about adding some logic related to api keys into it. If we want to change the env variable name or the config file key for some reason, we might easily forget to change it in this function.

So the best solution I found was to pass a callback that receives the resolved config from the config file, and let each command handle the duplication detection and print to stdout.

It can be argued that we can put the duplication detection inside `resolveConfigFromFileAndEnvironment` though, but I'm still afraid this will be hidden and not reflected in the name. 

What would you recommend?

Reply:
I've put something in the `validation` helper that seemed adequate, let me know what you think

Reply:
Yeah, I think that's adequate for now. At some point we may want to think about seeing if we can take parameter and environment resolution higher up in the tool, so it's always consistent across all of the commands, but I think that would be major version worthy....